### PR TITLE
[terra-dev-site] Migrate terra-dev-site to the terra-application mono repo.

### LINKED ISSUE #243: Migrate terra-dev-site v6 to terra-application monorepo
# Story

## Description

As a terra-engineer i want to migrate terra-dev-stie to the terra-application mono repo

## Motivation

So that I can make changes to terra-application and terra-dev-site at the same time and remove the circular dependency

## Definition of Done

We'll know we're done when terra-dev-site and it's tests are merged into main.

### PR DESCRIPTION
**Pull request by mjhenkes**
### Summary

This is largely just copying code. 

Files of note: package.json for terra-dev-site and anywhere where there were file paths mentioned (tests)

Closes #243

### Deployment Link

https://terra-applic-.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -112,12 +112,15 @@
...
-    "html-webpack-plugin": "^4.5.0",
+    "html-webpack-plugin": "^3.1.0",
     "jest": "^26.6.3",
     "lerna": "^3.16.4",
     "link-parent-bin": "^1.0.0",
     "loader-runner": "^4.1.0",
     "postcss": "^8.2.10",
+    "postcss-assets-webpack-plugin": "^4.0.0",
+    "postcss-custom-properties": "^10.0.0",
+    "postcss-loader": "^4.1.0",
```
Comment (mjhenkes) [inline package.json:123]: These are added to override the the older versions coming from terra-dev-site that we can't remove yet.

File: package.json [hunk 2]
```diff
@@ -112,12 +112,15 @@
     "enzyme-to-json": "^3.4.4",
     "eslint": "^7.19.0",
     "gh-pages": "^3.1.0",
-    "html-webpack-plugin": "^4.5.0",
+    "html-webpack-plugin": "^3.1.0",
```
Comment (mjhenkes) [inline package.json:115]: This wasn't used before, downgraded to match what is used in terra-dev-site currently.

File: dev-site-config/terra-dev-site-test-theme.scss [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    --terra-application-navigation-header-background-color: rgb(25, 166, 25);
+    --terra-application-navigation-header-background-image: radial-gradient(100% 70% at 50% 0%, rgba(70, 171, 83, 0.7), rgba(70, 171, 83, 0)),
+    radial-gradient(100% 100% at 0% 0%, rgb(25, 166, 32), rgba(25, 166, 32, 0)),
+    radial-gradient(100% 100% at 0% 100%, rgb(29, 153, 39), rgba(29, 153, 39, 0)),
+    radial-gradient(100% 100% at 100% 0%, rgb(95, 172, 105), rgba(95, 172, 105, 0)),
+    radial-gradient(100% 600% at 100% 100%, rgb(84, 160, 81), rgba(84, 160, 81, 0)),
+    radial-gradient(100% 70% at 50% 100%, rgb(81, 172, 86), rgba(81, 172, 86, 0)),
+    linear-gradient(rgb(25, 166, 25), rgba(25, 166, 25, 1));
+  }
+}
```
Comment (mjhenkes) [inline dev-site-config/terra-dev-site-test-theme.scss]:
```suggestion
}

```

File: dev-site-config/head.html [hunk 4]
```diff
@@ -0,0 +1 @@
+<script> console.log("head html script from a file") </script>
```
Comment (mjhenkes) [inline dev-site-config/head.html]:
```suggestion
<script> console.log("head html script from a file") </script>

```

**Review by tbiethman — APPROVED** [review]

File: packages/terra-dev-site/package.json [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    "terra-spacer": "^3.3.0",
+    "terra-status-view": "^4.10.0",
+    "terser-webpack-plugin": "^1.1.0",
+    "to-ast": "^1.0.0",
+    "webpack-merge": "^4.1.2"
+  },
+  "peerDependencies": {
+    "@babel/cli": "^7.4.4",
+    "core-js": "^3.1.3",
+    "@babel/core": "^7.4.4",
```
Comment (benbcai) [inline packages/terra-dev-site/package.json]:
```suggestion
    "@babel/core": "^7.4.4",
    "core-js": "^3.1.3",
```

**Review by benbcai — APPROVED** [review]

**Pull request merged**